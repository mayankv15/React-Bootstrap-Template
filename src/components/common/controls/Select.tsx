import {
  FormControl,
  InputLabel,
  MenuItem,
  Select as MuiSelect,
} from "@mui/material";
import React from "react";
import { useTranslation } from "react-i18next";
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles((theme) => ({
  customSelect: {
    "& .MuiSelect-root": {
      paddingTop: 10.5,
      paddingBottom: 10.5,
      color: theme.palette.primary.main,
    },
  },
}));

interface LabelProps {
  label?: string;
  required?: boolean;
  options?: any[];
  value?: string;
  disabled?: boolean;
  variant?: undefined | "standard" | "outlined" | "filled";
  onChange?: (event) => void;
}
const Select = (props: LabelProps) => {
  const {
    label,
    required,
    disabled,
    options,
    value,
    onChange,
    variant = "standard",
  } = props;
  const len = label?.length !== undefined ? label?.length : 1;
  const { t } = useTranslation();
  const handleChange = (event: any) => {
    if (onChange) {
      onChange(event);
    }
  };

  const classes = useStyles();
  return (
    <FormControl fullWidth>
      {label && (
        <InputLabel
          id="demo-simple-select-label"
          style={{ width: 10 * len, paddingLeft: 5 }}
        >
          {t(label)}
        </InputLabel>
      )}
      <MuiSelect
        labelId="demo-simple-select-label"
        id="demo-simple-select"
        variant={variant}
        className={classes.customSelect}
        value={value}
        required={required}
        disabled={disabled}
        onChange={handleChange}
        style={{ width: 15 * len, paddingTop: 15 }}
      >
        {options &&
          options.length > 0 &&
          options.map((option) => (
            <MenuItem key={option.value} value={option.value}>
              {t(option.title)}
            </MenuItem>
          ))}
      </MuiSelect>
    </FormControl>
  );
};
export default Select;
