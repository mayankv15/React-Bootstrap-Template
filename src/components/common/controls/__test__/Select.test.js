import React from "react";
import { render } from "@testing-library/react";
import Select from "../Select"

describe("Renders Select", () => {
    it("should render Select correctly", () => {
        const selectData = [
            {
                title: 'One',
                value: '1'
            },
            {
                title: 'Two',
                value: '2'
            },
            {
                title: 'Three',
                value: '3'
            },
        ]
        const component = render(<Select
            label="Dropdown options to be select from"
            options={selectData}
            variant="filled" />);
        expect(component).toMatchSnapshot();
    });
});
