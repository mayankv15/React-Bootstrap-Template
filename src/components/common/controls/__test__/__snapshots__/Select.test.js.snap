// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Select should render Select correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          id="demo-simple-select-label"
          style="width: 340px; padding-left: 5px;"
        >
          Dropdown options to be select from
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-customSelect-1 MuiInputBase-formControl"
          style="width: 510px; padding-top: 15px;"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
            id="demo-simple-select"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        id="demo-simple-select-label"
        style="width: 340px; padding-left: 5px;"
      >
        Dropdown options to be select from
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-customSelect-1 MuiInputBase-formControl"
        style="width: 510px; padding-top: 15px;"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
          id="demo-simple-select"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
